src/lib/replication/repl_dev/raft_state_machine.cpp

```diff
@@ -204,8 +204,49 @@ raft_buf_ptr_t RaftStateMachine::commit_ext(nuraft::state_machine::ext_op_params
 }
 
 void RaftStateMachine::commit_config(const ulong log_idx, raft_cluster_config_ptr_t& new_conf) {
+    // when reaching here, the config change log has already been committed, and the new config has been applied to the
+    // cluster
+
     RD_LOGD("Raft channel: Commit new cluster conf , log_idx = {}", log_idx);
-    // TODO:add more logic here if necessary
+    auto& servers_in_new_conf = new_conf->get_servers();
+    std::unordered_set< int32_t > server_ids_in_new_conf;
+    for (auto& server : servers_in_new_conf)
```

Comment:
```
for (auto& server : servers_in_new_conf) {
#ifdef _PRERELEASE
        server_ids_in_new_conf.insert(server->get_id());
#endif
    if (m_rd.server_id() == server->getid()) {
         m_rd.m_listener->on_leave_cluster(m_rd.group_id());
         m_rd.leave();
#ifndef _PRERELEASE
         break;
#endif
    }
}
```

Reply:
remove , I make the whole part off from release build. we will not handle leave cluster event here